File: study_assets_root/clock_drawing/analysis/extractImages.m
rootPath = '~/Dropbox/jatos/workshop_ASSC/examples/neglect';
jatosRoot = '/Applications/jatos-3.3.5_mac_java_old/study_assets_root/clock_drawing/';
dataPath = [rootPath filesep 'jatosResults'];

addpath(genpath([jatosRoot filesep 'analysis']))

dataFiles = dir([dataPath, filesep, '*.txt']);

if length(dataFiles)==1
    resultData = loadjson([dataPath, filesep, dataFiles(1).name]);
    nsubj = length(resultData);
    for subj = 1:nsubj
        %Remove the first 31 characters: "\"imgData=data:image/png;base64,"
        %And the last 3 characters: \""
        textToConvert = resultData{subj}{1}.image_data(32:end-3);
        %Save image
        base64decode(textToConvert, [dataPath filesep 'subj_' num2str(subj) 'clock_Matlab.png']);
    end
    
    
else
    for subj = 1:length(dataFiles)
        resultData = loadjson([dataPath, filesep, dataFiles(subj).name]);
        textToConvert = resultData{1}.image_data(32:end-3);
        base64decode(textToConvert, [dataPath filesep 'subj_' num2str(subj) 'clock_Matlab.png']);
    end
end

File: study_assets_root/clock_drawing/analysis/extractImages.R
rootPath = '~/Dropbox/jatos/workshop_ASSC/examples/neglect'
jatosRoot = '/Applications/jatos-3.3.5_mac_java_old/study_assets_root/clock_drawing/'
dataPath = paste(rootPath, "jatosResults", sep=.Platform$file.sep)


#load necessary libraries
library(jsonlite)
library(png)
library(Hmisc)
library(Rcurl)
source(paste(jatosRoot, 'analysis', 'getImg.R', sep= .Platform$file.sep))

# Take all exported text files from JATOS. 
# This can either be a single .txt file with all the study results exported at once
# Or a list of individual results
# They can come from either the component or the study results view, but they should contain data from the task component alone
dataFiles <- list.files(path = dataPath, pattern = ".txt")

# If results were exported individually
if (length(dataFiles)==1){
  #Read the whole thing in 
  fileName <- paste(dataPath,dataFiles[subj], sep= .Platform$file.sep)
  allJSONs <- readChar(fileName, file.info(fileName)$size)
  singleJSONs <- Hmisc::string.break.line(allJSONs)
  JSONdata <- singleJSONs 
  nSubj <- length(JSONdata[[1]])
} else {
  JSONdata <- dataFiles
  nSubj <- length(JSONdata)
}

for (subj in 1:nSubj){
  if (length(dataFiles)==1){
    resultData <- jsonlite::fromJSON(singleJSONs[[1]][subj])
  } else {
    resultData <- jsonlite::fromJSON(paste(dataPath,dataFiles[subj], sep=.Platform$file.sep))
  }
  #Remove the first 31 characters: "\"imgData=data:image/png;base64,"
  #And the last 3 characters: \""
  textToConvert <- substr(resultData$image_data, 32, nchar(resultData$image_data)-3)
  
  #Plot and save
  jpeg(paste0(dataPath, .Platform$file.sep, "subj_", subj, "clock_R.png"))
  plot(1:2, type='n', axes=FALSE, xlab="", ylab="")
  rasterImage(getImg(textToConvert), 1, 1, 2, 2, interpolate=TRUE)
  dev.off()
}


